### DISCUSSION TITLE
Hide column where position is in the middle

### DISCUSSION THREAD

**Discussion by calvinliem** [eclipse-birt/birt#1645] [Q&A] [answered]
Hello everyone,

So i have reportParameters let's call it isHideColumnCategory (boolean), if isHideColumnCategory is TRUE then i want to show only 2 column from 4 column. otherwise, i want it show all 4 column. The problem is i want to hide column where position is in the middle (column 2 and column 3).

This is what its look like if all of column is show (4 column total) when isHideColumnCategory is FALSE
![image](https://github.com/eclipse-birt/birt/assets/57037399/1d50873b-6d53-4fd1-8f14-a7e0fcf79f2e)

This is what its look like if im using Visibility properties to hide column number 2 and 3 when isHideColumnCategory is TRUE
![image](https://github.com/eclipse-birt/birt/assets/57037399/7a11e775-93e9-4939-9da1-0d41f6759b01)

2nd picture is showing empty column (column B and C) i dont want to show that empty column, is there anything to do?

I'm using BIRT Runtime 4.15 and designed with birt-report-designer-all-in-one-4.15.0-202403270652-win32.win32.x86_64

Thankyou before.

**Comment by hvbtup** [comment]
You can simply use a visibility expression for the column...

**Reply by calvinliem** [reply]
Thank you for the response, but i've used the feature and there's empty column in the middle. my goal is like this 
![image](https://github.com/eclipse-birt/birt/assets/57037399/f3cb83ab-2b72-4021-a36c-7f4bbb5a809d)

but if i use visibility expression for the column it will show like this
![image](https://github.com/eclipse-birt/birt/assets/57037399/d39892a4-5914-4c19-a61a-5047897bc0b1)

**Comment by hvbtup** [comment]
You have to set the visibility expression for the column itself, not for the column header cell or something.

**Reply by calvinliem** [reply]
i've set the visibility expression in this element sir, is that correct?
![image](https://github.com/eclipse-birt/birt/assets/57037399/8624bdd3-70e5-4e65-b898-bbe01ee3af1b)

**Reply by hvbtup** [reply, marked as answer]
No.
You need to set it for the column.
You cannot navigate to the column in the "Outline" tree.
You have to first select the table or grid (in your case: table myTable) in the layout editor. This will show handles for the _rows_ and for the _columns_. Then you click on the grey handle for the column as shown below:
![grafik](https://github.com/eclipse-birt/birt/assets/7644776/d6bfd596-18e3-4f72-b18a-51cec3064364)

This is the only way to set properties, e.g. the visibility expression, for the column.
![grafik](https://github.com/eclipse-birt/birt/assets/7644776/f27f49cf-be78-4e98-af4a-397abe45ba70)

**Reply by calvinliem** [reply]
Oh, i see. your solution is work for my case. thank you so much sir
